service/src/main/java/bio/terra/workspace/db/DbRetryUtils.java

```diff
@@ -0,0 +1,80 @@
+package bio.terra.workspace.db;
+
+import bio.terra.common.db.DatabaseRetryUtils;
+import java.time.Duration;
+import java.util.List;
+import org.springframework.dao.RecoverableDataAccessException;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+
+/**
+ * This class contains wrappers around JDBC methods which run them with retries. Only DB exceptions
+ * considered retryable ({@link RecoverableDataAccessException} and {@link
+ * org.springframework.dao.TransientDataAccessException}) are retried. See Spring documentation of
+ * these exceptions for more information:
+ * https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/dao/DataAccessException.html
+ *
+ * <p>Retries are with no delay, up to a certain maximum number of retries.
+ */
+public class DbRetryUtils {
+
+  public static int MAX_RETRIES = 10;
+
+  public static int update(
+      NamedParameterJdbcTemplate jdbcTemplate, String sql, MapSqlParameterSource params) {
+    try {
+      return DatabaseRetryUtils.executeAndRetry(
+          () -> jdbcTemplate.update(sql, params), Duration.ZERO, MAX_RETRIES);
+    } catch (InterruptedException e) {
+      throw new IllegalStateException(
```

Comment:
I don't think you should hide interrupted exception. Just because you are not sleeping does not mean that somewhere in the JDBC layers, there is not a wait. So I am not sure it can never happen.

Reply:
Done. I think throwing those as `InternalServerErrorException` from services (or letting Steps handle them) is the right level to surface them